src/umpire/strategy/DynamicPoolList.cpp

```diff
@@ -130,12 +130,37 @@ void DynamicPoolList::coalesce() noexcept
 
 PoolCoalesceHeuristic<DynamicPoolList> DynamicPoolList::blocks_releasable(std::size_t nblocks)
 {
-  return [=](const strategy::DynamicPoolList& pool) {
-    return pool.getReleasableBlocks() > nblocks ? pool.getActualSize() : 0;
-  };
+  return blocks_releasable(nblocks, false);
+}
+
+PoolCoalesceHeuristic<DynamicPoolList> DynamicPoolList::blocks_releasable_hwm(std::size_t nblocks)
+{
+  return blocks_releasable(nblocks, true);
+}
+
+PoolCoalesceHeuristic<DynamicPoolList> DynamicPoolList::blocks_releasable(std::size_t nblocks, bool hwm_switch)
+{
+  if (hwm_switch)
+    return [=](const strategy::DynamicPoolList& pool) {
+      return pool.getReleasableBlocks() > nblocks ? pool.getHighWatermark() : 0;
+    };
+  else
+    return [=](const strategy::DynamicPoolList& pool) {
+      return pool.getReleasableBlocks() > nblocks ? pool.getActualSize() : 0;
+    };
 }
 
 PoolCoalesceHeuristic<DynamicPoolList> DynamicPoolList::percent_releasable(int percentage)
+{
+  return percent_releasable(percentage, false);
+}
+
+PoolCoalesceHeuristic<DynamicPoolList> DynamicPoolList::percent_releasable_hwm(int percentage)
+{
+  return percent_releasable(percentage, true);
+}
+
+PoolCoalesceHeuristic<DynamicPoolList> DynamicPoolList::percent_releasable(int percentage, bool hwm_switch)
```

Comment:
Can we rename `hwm_switch` to something like `constrain_to_hwm` or `coalesce_to_hwm` or `resize_to_hwm`?  I'm not sure I understand what `hwm_switch` does by its name.

Reply:
That is a good point! I like `coalesce_to_hwm` or `resize_to_hwm`.

Reply:
@davidbeckingsale any preference here?

Reply:
`resize_to_hwm`